src/python/pants/backend/python/util_rules/pex.py

```diff
@@ -403,19 +402,29 @@ class _BuildPexRequirementsSetup:
 
 @rule_helper
 async def _setup_pex_requirements(
-    request: PexRequest, python_repos: PythonRepos, python_setup: PythonSetup
+    request: PexRequest, python_setup: PythonSetup
 ) -> _BuildPexRequirementsSetup:
-    pex_lock_resolver_args = list(python_repos.pex_args)
-    pip_resolver_args = [*python_repos.pex_args, "--resolver-version", "pip-2020-resolver"]
+    resolve_name: str | None
+    if isinstance(request.requirements, EntireLockfile):
+        resolve_name = request.requirements.lockfile.resolve_name
+    elif isinstance(request.requirements.from_superset, LoadedLockfile):
+        resolve_name = request.requirements.from_superset.original_lockfile.resolve_name
+    else:
+        # This implies that, currently, per-resolve options are only configurable for lockfiles. If
+        # no resolve is specified, we will still load options that apply to every resolve, like
+        # `[python-repos]`.
+        resolve_name = None
```

Comment:
Once `[python-repos]` is migrated, we will have a way to set the `resolve_name` when you're not using a lockfile. (`PexRequirements` will have you set the value)